external/dhcpc/dhcpc.c

```diff
@@ -332,7 +335,14 @@ static uint8_t dhcpc_parseoptions(struct dhcpc_state *presult, uint8_t *optptr,
 		case DHCP_OPTION_DNS_SERVER:
 			/* Get the DNS server address in network order */
 
-			memcpy(&presult->dnsaddr.s_addr, optptr + 2, 4);
+			total_dns = (*(optptr+1))/4;
```

Comment:
do you like to add some comments for below code here?

Reply:
added.